The error "No such file or directory: '/var/run/awssmatoken'" occurs in Lambda.

https://github.com/aws/aws-secretsmanager-agent
Following the steps above, I performed the following:

1. Cloned this repository in Cloud9.
2. Followed the instructions for "Step 1: RPM-based systems" at the root of the repository.
3. Executed the instructions for "Step 2: AWS Lambda."

## Issue
I successfully executed all commands in Step 3-2, and the Layer was properly added to the Lambda. 
However, when running the Lambda function, the following error occurs:

`
[ERROR] Exception: Error: [Errno 2] No such file or directory: '/var/run/awssmatoken'
Traceback (most recent call last):
  File "/var/task/lambda_function.py", line 25, in lambda_handler
    raise Exception(f"Error: {e}")
`

## What I Tried

- Set the environment variable AWS_TOKEN for Lambda.
  - I left the value empty because I was unsure what to set.
- Implemented retry logic.

## My Code

```
import requests
import json
import time

# Function that fetches the secret from Secrets Manager Agent for the provided secret id. 
def lambda_handler(event, context):

    for attempt in range(6):
        # Construct the URL for the GET request
        url = f"http://localhost:2773/secretsmanager/get?secretId=test"

        try:
            # Get the SSRF token from the token file
            with open('/var/run/awssmatoken') as fp:
                token = fp.read() 

            headers = {
                "X-Aws-Parameters-Secrets-Token": token.strip()
            }
        
        except Exception as e:
            if attempt < 6 - 1:
                time.sleep(10)
            else:
                raise Exception(f"Error: {e}")

    try:
        # Send the GET request with headers
        response = requests.get(url, headers=headers)

        # Check if the request was successful
        if response.status_code == 200:
            # Return the secret value
            return response.text
        else:
            # Handle error cases
            raise Exception(f"Status code {response.status_code} - {response.text}")

    except Exception as e:
        # Handle network errors
        raise Exception(f"Error: {e}")
```

How can I resolve this error?

Related Issue: https://github.com/aws/aws-secretsmanager-agent/issues/52

> did you create the lambda layer in the second step and apply it to your function?
Yes, as stated in the README, the registration to the Lambda Layer was successfully completed by executing the following command.

```
# Publish the layer version
LAYER_VERSION_ARN=$(aws lambda publish-layer-version \
    --layer-name secrets-manager-agent-extension \
    --zip-file "fileb://secrets-manager-agent-extension.zip" | jq -r '.LayerVersionArn')

# Attach the layer version to the Lambda function
aws lambda update-function-configuration \
    --function-name $LAMBDA_ARN \
    --layers "$LAYER_VERSION_ARN"
```

From the logs, it was also confirmed that the Secrets Manager Agent process is being executed during Lambda's runtime.

```
[secrets-manager-agent-extension.sh] Initialization
[secrets-manager-agent-extension.sh] Registering at http://169.254.100.1:9001/2020-01-01/extension/register
[secrets-manager-agent-extension.sh] Registration response: {"functionName":"SecretsManagerAgent","functionVersion":"$LATEST","handler":"lambda_function.lambda_handler"} with EXTENSION_ID  fbce693c-24d3-4b02-be97-946c13610489
[secrets-manager-agent-extension.sh] Waiting for event. Get /next event from http://169.254.100.1:9001/2020-01-01/extension/event/next
EXTENSION	Name: secrets-manager-agent-extension.sh	State: Ready	Events: [INVOKE, SHUTDOWN]
START RequestId: 4333ef35-014d-46bd-87ba-d316eada2cf6 Version: $LATEST
[extension: secrets-manager-agent-extension.sh] Received INVOKE event.
[secrets-manager-agent-extension.sh] Starting Secrets Manager Agent.
[secrets-manager-agent-extension.sh] Checking if the Agent is serving requests.
LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
[ERROR] FileNotFoundError: [Errno 2] No such file or directory: '/var/run/awssmatoken'
Traceback (most recent call last):
  File "/var/task/lambda_function.py", line 10, in lambda_handler
    with open('/var/run/awssmatoken') as fp:
[secrets-manager-agent-extension.sh] Agent has not started yet, retrying in 100 milliseconds...
Agent/1.0.1 listening on http://127.0.0.1:2773
[secrets-manager-agent-extension.sh] Agent has started.
[secrets-manager-agent-extension.sh] Waiting for event. Get /next event from http://169.254.100.1:9001/2020-01-01/extension/event/next
END RequestId: 4333ef35-014d-46bd-87ba-d316eada2cf6
REPORT RequestId: 4333ef35-014d-46bd-87ba-d316eada2cf6	Duration: 126.03 ms	Billed Duration: 127 ms	Memory Size: 5000 MB	Max Memory Used: 74 MB	Init Duration: 463.44 ms	
```

However, ultimately, the error mentioned in the title occurred.

--- a/README.md
+++ b/README.md
@@ -235,9 +235,7 @@ The following instructions show how to get a secret named *MyTest* by using the 
 
 **To create a Lambda extension that packages the Secrets Manager Agent**
 
-1. Create a Python Lambda function that reads the SSRF token from environment variable `AWS_TOKEN`, and queries `http://localhost:2773/secretsmanager/get?secretId=MyTest` to get the secret. Be sure to specify environment variable `AWS_TOKEN` for your lambda, and additionally, implement retry logic in your application code to accommodate delays in initialization and registration of the Lambda extension\.
-
-1. From the root of the Secrets Manager Agent code package, run the following commands to test the Lambda extension\.
+1. Package the agent as a layer. From the root of the Secrets Manager Agent code package, run the following example commands\:
 
    ```sh
    AWS_ACCOUNT_ID=<AWS_ACCOUNT_ID>
@@ -250,7 +248,7 @@ The following instructions show how to get a secret named *MyTest* by using the 
    mkdir -p ./bin
    cp ./target/x86_64-unknown-linux-gnu/release/aws_secretsmanager_agent ./bin/secrets-manager-agent
    
-   # Copy the `secrets-manager-agent-extension.sh` script into the `extensions` folder.
+   # Copy the `secrets-manager-agent-extension.sh` example script into the `extensions` folder.
    mkdir -p ./extensions
    cp aws_secretsmanager_agent/examples/example-lambda-extension/secrets-manager-agent-extension.sh ./extensions
    
@@ -261,14 +259,22 @@ The following instructions show how to get a secret named *MyTest* by using the 
    LAYER_VERSION_ARN=$(aws lambda publish-layer-version \
        --layer-name secrets-manager-agent-extension \
        --zip-file "fileb://secrets-manager-agent-extension.zip" | jq -r '.LayerVersionArn')
-   
+   ```
+
+2. The default configuration of the agent will automatically set the SSRF token to the value set in the pre-set `AWS_SESSION_TOKEN` or `AWS_CONTAINER_AUTHORIZATION_TOKEN` environment variables (the latter variable for Java-based Lambda functions with SnapStart enabled). Alternatively, you can define the `AWS_TOKEN` environment variable with an arbitrary value for your Lambda function instead as this variable takes precedence over the other two. If you choose to use the `AWS_TOKEN` environment variable, you must set that environment variable with a `lambda:UpdateFunctionConfiguration` call\.
+
+
+3. Attach the layer version  to your Lambda function:
+   ```sh
    # Attach the layer version to the Lambda function
    aws lambda update-function-configuration \
        --function-name $LAMBDA_ARN \
        --layers "$LAYER_VERSION_ARN"
    ```
+4. Update your Lambda function to query `http://localhost:2773/secretsmanager/get?secretId=MyTest` with the `X-Aws-Parameters-Secrets-Token` header value set to the value of the SSRF token sourced from one the environment variables mentioned above to retrieve the secret. Be sure to implement retry logic in your application code to accommodate delays in initialization and registration of the Lambda extension\.
 
-1. Invoke the Lambda function to verify that the secret is being correctly fetched\. 
+
+5. Invoke the Lambda function to verify that the secret is being correctly fetched\. 
 
 ------
 
@@ -353,7 +359,7 @@ The following list shows the options you can configure for the Secrets Manager A
 + **ttl\_seconds** – The TTL in seconds for the cached items, in the range 1 to 3600\. The default is 300\. This setting is not used if the cache size is 0\. 
 + **cache\_size** – The maximum number of secrets that can be stored in the cache, in the range 0 to 1000\. 0 indicates that there is no caching\. The default is 1000\. 
 + **ssrf\_headers** – A list of header names the Secrets Manager Agent checks for the SSRF token\. The default is "X\-Aws\-Parameters\-Secrets\-Token, X\-Vault\-Token"\. 
-+ **ssrf\_env\_variables** – A list of environment variable names the Secrets Manager Agent checks for the SSRF token\. The environment variable can contain the token or a reference to the token file as in: `AWS_TOKEN=file:///var/run/awssmatoken`\. The default is "AWS\_TOKEN, AWS\_SESSION\_TOKEN"\.
++ **ssrf\_env\_variables** – A list of environment variable names the Secrets Manager Agent checks in sequential order for the SSRF token\. The environment variable can contain the token or a reference to the token file as in: `AWS_TOKEN=file:///var/run/awssmatoken`\. The default is "AWS\_TOKEN, AWS\_SESSION\_TOKEN, AWS\_CONTAINER\_AUTHORIZATION\_TOKEN\".
 + **path\_prefix** – The URI prefix used to determine if the request is a path based request\. The default is "/v1/"\.
 + **max\_conn** – The maximum number of connections from HTTP clients that the Secrets Manager Agent allows, in the range 1 to 1000\. The default is 800\.
 

